fix: Reset form on cancel button click in schedule template sheet

siva222003 opened the pull request:
## Proposed Changes

- Fixes #12224 
- Added form reset on cancel to prevent stale data when reopening the schedule template form.

[Screencast from 2025-05-12 23-00-52.webm](https://github.com/user-attachments/assets/aefc79cf-883f-4e9e-89c3-71c08217647a)


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Bug Fixes**
  - Improved the "Cancel" button in the schedule template creation sheet to clear form fields when clicked, ensuring the form resets to its default state.
  - Enhanced cancel buttons across multiple forms to reset input fields and UI states upon cancellation.
  - Updated form sheets to separate form reset logic from sheet open/close events, triggering resets only on user-initiated cancel actions.
  - Added clearing of search inputs and selections when closing relevant sheets.
  - Ensured device selection is cleared when opening or closing the associate device sheet.
- **New Features**
  - Introduced explicit cancel buttons alongside submit buttons in various forms for clearer user control.
  - Improved button layouts with grouped cancel and submit buttons for better usability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

siva222003 committed:
fix: Reset form on cancel button click in schedule template sheet

rithviknishad reviewed (changes requested):
Please do this for other such form sheets as well (such as schedule exception) along with this PR wherever applicable!

<img width="725" alt="image" src="https://github.com/user-attachments/assets/6f76db5d-c636-450a-af2a-9a44b8306f90" />

Jacobjeevan commented on src/pages/Scheduling/components/CreateScheduleTemplateSheet.tsx:
```diff
@@ -700,7 +700,12 @@ export default function CreateScheduleTemplateSheet({
+                    variant="outline"
+                    type="button"
+                    disabled={isPending}
+                    onClick={() => form.reset()}
```
handle within sheet state instead (onOpenChange), since user can close by clicking elsewhere on screen.

rithviknishad replied:
Let's not do this, this was intentionally not done so that sheet's state can be restored.

What if user accidentally closes the sheet? They can restore it.

But if they voluntarily clicked cancel, form state would be discarded.

ref: https://github.com/ohcnetwork/care_fe/issues/12224#issuecomment-2870923191

siva222003 replied:
@Jacobjeevan But here the form should reset only when we click `cancel`, right?

siva222003 replied:
@rithviknishad There are some sheets that already manage state using `onOpenChange` to reset the form. So, do you want me to keep them as they are, or should they also reset only when `cancel` is clicked?

And also, sheets like `CreateEncounter`, which don’t have a cancel button, are also using this approach.

- My idea is to remove the reset logic from `onOpenChange` for the sheets that have a Cancel button and reset the form only when Cancel is clicked.

- And for the sheets that don’t have a cancel option, I’ll keep the current behavior as is ,resetting on `onOpenChange` (or remove it).

What do you think?

Jacobjeevan replied:
> @rithviknishad There are some sheets that already manage state using `onOpenChange` to reset the form. So, do you want me to keep them as they are, or should they also reset only when `cancel` is clicked?
> 
> And also, sheets like `CreateEncounter`, which don’t have a cancel button, are also using this approach.
> 
>     * My idea is to remove the reset logic from `onOpenChange` for the sheets that have a Cancel button and reset the form only when Cancel is clicked.
> 
>     * And for the sheets that don’t have a cancel option, I’ll keep the current behavior as is ,resetting on `onOpenChange` (or remove it).
> 
> 
> What do you think?

(1) Yep
(2) This seems fine, but check if the reset is expensive for the user. All form sheets should have a cancel button, so let's add it if it's missing. For other things, we can keep the behavior as is 🤔

rithviknishad replied:
@siva222003 could you update the status and give an updated ETA on this?

siva222003 replied:
@rithviknishad Sorry, I was occupied with the GSoC task and couldn't complete this. I'll make the changes by EOD tomorrow.
I've also updated the ETA in the issue.

rithviknishad replied:
Cool, no worries! It's okay to take time, just make sure you add a new comment every time you think you'd need more time to get things wrapped...

siva222003 replied:
yeah, I'll keep that in mind 🙂

siva222003 committed:
add reset on cancel for multiple sheets

siva222003 commented:
@rithviknishad @Jacobjeevan
I have added the reset on the sheets where I felt it was needed.  
Also, I added a `Cancel` button in two sheets to handle the state reset when clicking `Cancel` instead of relying on `onOpenChange`.

Please review and let me know if any changes are needed.

rithviknishad commented on src/pages/Encounters/AssociateDeviceSheet.tsx:
```diff
@@ -57,7 +57,13 @@ export default function AssociateDeviceSheet({
-    <Sheet open={open} onOpenChange={setOpen}>
+    <Sheet
+      open={open}
+      onOpenChange={() => {
+        setOpen((prev) => !prev);
+        setSelectedDevice(null);
+      }}
```
```suggestion
      onOpenChange={(open) => {
        setOpen(open);
        setSelectedDevice(null);
      }}
```

rithviknishad reviewed (approved):
minor thing

rithviknishad committed:
Update src/pages/Encounters/AssociateDeviceSheet.tsx

nihal467 commented:
LGTM
